src/app.ts

```diff
@@ -13,14 +13,46 @@ interface TemperatureSummary {
   average: number
 }
 
+const formatDate = (date: Date): string => date.toLocaleDateString()
+
+const temperatureData: Map<string, Map<string, number[]>> = new Map()
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  readings.forEach(({ city, temperature, time }) => {
+    const formattedDate = formatDate(time)
+
+    if (!temperatureData.has(city)) {
+      temperatureData.set(city, new Map())
+    }
+
+    const cityMap = temperatureData.get(city)
+
+    if (!cityMap?.has(formattedDate)) {
+      cityMap?.set(formattedDate, [])
+    }
+
+    const cityMapDate = cityMap?.get(formattedDate)
+    cityMapDate?.push(temperature)
+  })
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
+  const cityData = temperatureData.get(city)
+
+  if (!cityData) return null
```

Comment:
To improve readability, we use braces in all if statements, even if they are only one line.